KoPrivateGPT/retrieval/base.py

```diff
@@ -1,14 +1,21 @@
+import json
 from abc import ABC, abstractmethod
-from typing import List, Union
+from typing import List, Union, Any, Dict
 from uuid import UUID
 
-from KoPrivateGPT.DB.base import BaseDB
+from KoPrivateGPT.DB import MongoDB, PickleDB
 from KoPrivateGPT.schema import Passage
+from KoPrivateGPT.utils.linker import RedisDBSingleton
 
 
 class BaseRetrieval(ABC):
+    def __init__(self):
+        self.mongo_db = None
```

Comment:
What if we add new DB type? (like postgreSQL). We have to make all of this kind of value, and custom function for that like fetch_mongo_data. It is so bad structure. 

I proposed method below. 

First, you can build any instance with using only db_type and db_path. With db_type, you can create the right instance. (like selector in pipeline module)
db_path(dict) will be parameter for initializing that DB instance. 

Second, you can build stack inside BaseRetrieval.  That stack can store DB instances. When you push to stack, you must check that DB instance is not duplicated. If duplicate, just use already created one. 

Third, you can check duplicate db instance using only DBOrigin. Build __eq__ func at DBOrigin.